service/frontend/workflow_handler.go

```diff
@@ -4145,9 +4147,10 @@ func (wh *WorkflowHandler) unaliasStartWorkflowExecutionRequestSearchAttributes(
 	}
 
 	// Shallow copy request and replace SearchAttributes fields only.
-	newRequest := *request
+	// FIXME actually shallow copy
+	newRequest := common.CloneProto(request)
 	newRequest.SearchAttributes = unaliasedSas
-	return &newRequest, nil
+	return newRequest, nil
```

Comment:
also, definitely not in this PR, but I think this pattern of replacing the SA field in SWERequest and other messages is error-prone (see the recent bug that Roey fixed). we should use the type system to differentiate between aliased and unaliased names. (which I would call external and internal since "alias" could be read both ways.) all user-facing api messages should have only external names, and internal names should be present in internal structs only. ideally confined to visibility, but that may be hard.

anyway, that's a big refactoring that can happen later.